### PR TITLE
common_properties moved to BaseScheduler

### PR DESCRIPTION
**Pull request by ImmortalRabbit**
@wtgee Sorry for recreating many times one Pull Request, here is my solution, I just have moved common_properties to BaseScheduler, that's how I basically understood this issue. #750

### CODE REVIEW

File: pocs/scheduler/dispatch.py [hunk 1]
```diff
@@ -46,11 +46,7 @@ def get_observation(self, time=None, show_all=False, reread_fields_file=False):
         valid_obs = {obs: 1.0 for obs in self.observations}
         best_obs = []
 
-        common_properties = {
-            'end_of_night': self.observer.tonight(time=time, horizon=-18 * u.degree)[-1],
-            'moon': get_moon(time, self.observer.location),
```
Comment (wtgee) [inline pocs/scheduler/dispatch.py]: Can you remove the `get_moon` import from the top of this file as it is no longer used.

File: pocs/scheduler/scheduler.py [hunk 2]
```diff
@@ -59,6 +60,7 @@ def __init__(self, observer, fields_list=None, fields_file=None,
             self.logger.debug("Reading initial set of fields")
             self.read_field_list()
 
+        self.common_properties = None
```
Comment (wtgee) [inline pocs/scheduler/scheduler.py]:
```suggestion
        # Items common to each observation that shouldn't be computed each time.
        self.common_properties = None
```

Add a comment describing what this is.

File: pocs/scheduler/scheduler.py [hunk 3]
```diff
@@ -274,6 +276,14 @@ def read_field_list(self):
                 except Exception as e:
                     self.logger.warning("Error adding field: {}", e)
 
+    def set_common_properties(self, time):
+
+        self.common_properties = {
```
Comment (wtgee) [inline pocs/scheduler/scheduler.py]: Not really part of this PR, but since you are right here can you change this horizon to pull from the config rather than be hard-coded?

```python
horizon_limit = self.config['location'].get('observe_horizon', -18 * u.degree)      
self.common_properties = {
    'end_of_night': self.observer.tonight(time=time, horizon=horizon_limit)[-1],
    'moon': get_moon(time, self.observer.location),
    'observed_list': self.observed_list
}
```

**Comment by ImmortalRabbit** [comment]
@wtgee Could you check my commit and did I get all your points?
